Quirk news.ycombinator to skip TextAutoSizing

vitorroriz opened the pull request:
#### 84ae355619354ee1bfa7daaa1fc95565a6726be3
<pre>
Quirk news.ycombinator to skip TextAutoSizing
<a href="https://rdar.apple.com/127246368">rdar://127246368</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=275221">https://bugs.webkit.org/show_bug.cgi?id=275221</a>

Reviewed by Sammy Gill.

We won&apos;t try to adjust computed font sizes if (visibleWidth &gt;= width())
at RenderBlockFlow::adjustComputedFontSizes.

On the first load, RenderBlockFlow::width is calculated to a value that
allows TextAutoSizing to set new computed font sizes for the text nodes
on the page. Width is much larger than visibleWidth, ultimately coming from
WebPageProxy::textAutosizingWidth which returns WebCore::screenSize().width().

At refresh,RenderBlockFlow::width returns a value that is actually equal
to visibleWidth (and screenSize().width()). There seems to depend on
timing. I wasn&apos;t able to reproduce that when simulator CPU load was too
high.

Since the page where the bug manifests itself already has responsive CSS
styling, we can quirk TextAutoSizing to skip adjusting for it, at least
until we figure out why we are calculating RenderBlockFlow width inconsistently:
<a href="https://bugs.webkit.org/show_bug.cgi?id=275223">https://bugs.webkit.org/show_bug.cgi?id=275223</a>

* Source/WebCore/page/LocalFrameViewLayoutContext.cpp:
(WebCore::LocalFrameViewLayoutContext::applyTextSizingIfNeeded):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::skipTextAutoSizing const):
* Source/WebCore/page/Quirks.h:

Canonical link: <a href="https://commits.webkit.org/279788@main">https://commits.webkit.org/279788@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1288a055a2bacd3805f61faf2458b46bb917860f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54560 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33979 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7132 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57841 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5294 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56862 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41518 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5296 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44193 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3560 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56655 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32126 "Found 60 new test failures: accessibility/accessibility-node-reparent.html accessibility/ancestor-computation.html accessibility/announcement-notification.html accessibility/aria-braillelabel.html accessibility/aria-brailleroledescription.html accessibility/aria-checked-mixed-value.html accessibility/aria-current-state-changed-notification.html accessibility/aria-current.html accessibility/aria-describedby-on-input.html accessibility/aria-description.html ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47254 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25316 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28922 "Found 60 new test failures: imported/w3c/web-platform-tests/FileAPI/Blob-methods-from-detached-frame.html imported/w3c/web-platform-tests/FileAPI/FileReader/Progress_event_bubbles_cancelable.html imported/w3c/web-platform-tests/FileAPI/FileReader/workers.html imported/w3c/web-platform-tests/FileAPI/FileReaderSync.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-array-buffer.any.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-array-buffer.any.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-bytes.any.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-bytes.any.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-dom.window.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-endings.html ... (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4584 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3434 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50681 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4800 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59431 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/29793 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/4949 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/51614 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30943 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47340 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/50991 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11920 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/31921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/30731 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

stwrt reviewed (commented):
Can you please include the bugzilla link and the commit message.

karlcow commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1879,5 +1879,13 @@ bool Quirks::needsLaxSameSiteCookieQuirk() const
 
     return *m_needsLaxSameSiteCookieQuirk;
 }
+#if ENABLE(TEXT_AUTOSIZING)
```
Could you add the radar and/or bugs.webkit.org?
See elsewhere in the code for example.

sammygill commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1871,5 +1871,14 @@ bool Quirks::needsLaxSameSiteCookieQuirk(const URL& requestURL) const
 }
+#if ENABLE(TEXT_AUTOSIZING)
+// rdar://127246368
+bool Quirks::skipTextAutoSizing() const
```
Seems like the quirks seem to follow a naming convention of `needsX` or `shouldX` so I feel like we should do the same

vitorroriz committed:
Quirk news.ycombinator to skip TextAutoSizing
rdar://127246368
https://bugs.webkit.org/show_bug.cgi?id=275221

Reviewed by Sammy Gill.

We won't try to adjust computed font sizes if (visibleWidth >= width())
at RenderBlockFlow::adjustComputedFontSizes.

On the first load, RenderBlockFlow::width is calculated to a value that
allows TextAutoSizing to set new computed font sizes for the text nodes
on the page. Width is much larger than visibleWidth, ultimately coming from
WebPageProxy::textAutosizingWidth which returns WebCore::screenSize().width().

At refresh,RenderBlockFlow::width returns a value that is actually equal
to visibleWidth (and screenSize().width()). There seems to depend on
timing. I wasn't able to reproduce that when simulator CPU load was too
high.

Since the page where the bug manifests itself already has responsive CSS
styling, we can quirk TextAutoSizing to skip adjusting for it, at least
until we figure out why we are calculating RenderBlockFlow width inconsistently:
https://bugs.webkit.org/show_bug.cgi?id=275223

* Source/WebCore/page/LocalFrameViewLayoutContext.cpp:
(WebCore::LocalFrameViewLayoutContext::applyTextSizingIfNeeded):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::skipTextAutoSizing const):
* Source/WebCore/page/Quirks.h:

Canonical link: https://commits.webkit.org/279788@main
